src/js/player.js

```diff
@@ -2076,7 +2076,15 @@ class Player extends Component {
    * @method currentSources
    */
   currentSources() {
-    return this.cache_.sources || [this.currentSource()];
+    const source = this.currentSource();
+    const sources = [];
+
+    // assume `{}` or `{ src }`
+    if (Object.keys(source).length !== 0) {
+      sources.push(source);
+    }
+
+    return this.cache_.sources || sources;
```

Comment:
I'm not sure this is a big concern. It's fine to leave as is and we can always profile and increase performance in the future.